asapdiscovery-alchemy/asapdiscovery/alchemy/schema/fec.py

```diff
@@ -359,11 +406,35 @@ def to_alchemical_network(self) -> openfe.AlchemicalNetwork:
                     sys_b_dict, name=f"{mapping.componentB.name}_{leg}"
                 )
 
+                ## make the OpenFE protocol for this edge
+                # double the simulation time if requested
+                protocol_copy = self.copy(deep=True)
+                if (
+                    self.adaptive_settings.adaptive_sampling
+                    and scorer(mapping)
+                    < self.adaptive_settings.adaptive_sampling_threshold
+                ):
+                    protocol_copy.simulation_settings.production_length *= (
+                        self.adaptive_settings.adaptive_sampling_multiplier
+                    )
+                # adjust solvent padding per phase if requested
+                if self.adaptive_settings.adaptive_solvent_padding:
+                    if leg == "solvent":
+                        protocol_copy.solvation_settings.solvent_padding = (
+                            self.adaptive_settings.solvent_padding_solvated
+                        )
+                    else:
+                        protocol_copy.solvation_settings.solvent_padding = (
+                            self.adaptive_settings.solvent_padding_complex
+                        )
+                protocol_openfe = protocol_copy.to_openfe_protocol()
+
+                # set up the transformation
```

Comment:
Lets move all of this logic to a function on the `AdaptiveSettings` class that will make this section easier to read as this will get more complicated as the class evolves.